src/services/pokemonService.js

```diff
@@ -0,0 +1,120 @@
+import axios from 'axios'
+
+class PokemonService {
+  async getOnePokemon(url) {
+    return axios
+      .get(url)
+      .then(response => response.data)
+      .then(data => {
+        // assignment data.
+        let result = {}
+        result.id = data?.id
+        result.name = data?.name
+        result.types = data?.types
+        result.frontDefault = data?.sprites.front_default
+        const types = Object.values(data?.types).map(type => type.type.name)
+        result.types = types.join(', ') + '.'
+        result.baseExperience = data?.base_experience
+        result.weight = data?.weight
+        result.height = data?.height
+
+        // Refactor object key.
+        result.abilities = data?.abilities.map(item => {
+          item = {
+            ...item,
+            isHidden: item.is_hidden
+          }
+          delete item.is_hidden
+          return item
+        })
+
+        // Refactor object key.
+        result.stats = data?.stats.map(item => {
+          item = {
+            ...item,
+            baseStat: item.base_stat
+          }
+          delete item.base_stat
+          return item
+        })
+
+        result.speciesUrl = data?.species.url
+
+        // set data for state.
+        return result
+      })
+  }
+
+  async getEvolutionChain(url) {
+    const species = await axios.get(`${url}`)
+    const resEvolutionChain = await axios.get(
+      `${species.data.evolution_chain?.url}`
+    )
+    const data = resEvolutionChain.data?.chain
+    const speciesOne = data.species.name
+    const urlOfSpeciesOne = data.species.url
+
+    const speciesTwo = data.evolves_to[0]?.species.name
+    const urlOfSpeciesTwo = data.evolves_to[0]?.species.url
+
+    const speciesThree = data.evolves_to[0]?.evolves_to[0]?.species.name
+    const urlOfSpeciesThree = data.evolves_to[0]?.evolves_to[0]?.species.url
+    return [
+      { name: speciesOne, url: urlOfSpeciesOne },
+      { name: speciesTwo, url: urlOfSpeciesTwo },
+      { name: speciesThree, url: urlOfSpeciesThree }
+    ]
+  }
+
+  async getAbilityOfPokemon(url) {
+    // get Ability Of Pokemon.
+    return (
+      axios
+        .get(`${url}`)
```

Comment:
I don't catch exeption for the old code so if url are undefined app will stop running, i wrap url by template string to make this just like a wrong url, it will show at console but not stop app.
**i will fix that**